libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowContainer.m

```diff
@@ -0,0 +1,289 @@
+        if (currentViewController!=viewController)
+            [currentViewController presentViewController:viewController animated:NO completion:completion];
+    }
+    else {
+        self.window.rootViewController = viewController;
+        if (completion)
+            completion();
+    }
+    
+    [self enumerateDelegates:^(id<SFSDKWindowContainerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowDidPushViewController:controller:)]) {
+            [delegate windowDidPushViewController:weakSelf controller:viewController];
+        }
+    }];
+    
+}
+
+- (void)popViewController:(UIViewController *)viewController animated:(BOOL)flag completion:(void (^)(void))completion {
+    
+    __weak typeof(self) weakSelf = self;
+    if (![NSThread isMainThread]) {
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            [weakSelf popViewController:viewController animated:flag completion:completion];
+        });
+        return;
+    }
+    
+    [self enumerateDelegates:^(id<SFSDKWindowContainerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowWillPopViewController:controller:)]) {
+            [delegate windowWillPushViewController:weakSelf controller:viewController];
+        }
+    }];
+    
+    UIViewController *currentViewController = self.window.rootViewController;
+    if (viewController && currentViewController != viewController) {
+        // look for controller
+        while (currentViewController && currentViewController.presentedViewController!=viewController) {
+            currentViewController = currentViewController.presentedViewController;
+        }
+        // if controller is found dismiss the view, invoke delegates && restore Alerts if required.
+        if (viewController == currentViewController.presentedViewController) {
+            [self dismissPresentedViewController:currentViewController dismissViewControllerAnimated:flag completion:completion];
+        }
+    } else {
+        self.window.rootViewController = nil;
+        if (completion)
+            completion();
+    }
+    
+    [self enumerateDelegates:^(id<SFSDKWindowContainerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowDidPopViewController:controller:)]) {
+            [delegate windowWillPushViewController:weakSelf controller:viewController];
+        }
+    }];
+    
+}
+
+- (void)makeKeyVisible {
+    __weak typeof(self) weakSelf = self;
+    if (![NSThread isMainThread]) {
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            [weakSelf makeKeyVisible];
+        });
+        return;
+    }
+    [self enumerateDelegates:^(id<SFSDKWindowContainerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowWillMakeKeyVisible:)]) {
+            [delegate windowWillMakeKeyVisible:weakSelf];
+        }
+    }];
+    
+    self.window.windowLevel= fabs(self.window.windowLevel);
+    [self.window setHidden:NO];
```

Comment:
Toggling on `hidden` precludes the ability to animate these changes in and out.  I'd recommend using `alpha` to drive the visibility of the window, which allows for animation.